x-pack/plugins/fleet/server/services/epm/archive/storage.ts

```diff
@@ -147,89 +139,32 @@ export async function getAsset(opts: {
 
   return storedAsset;
 }
+
 export const getEsPackage = async (
   pkgName: string,
   pkgVersion: string,
-  packageAssets: SavedObjectsBulkGetObject[],
+  packageAssets: PackageAssetReference[],
   savedObjectsClient: SavedObjectsClientContract
 ) => {
-  const pkgKey = pkgToPkgKey({ name: pkgName, version: pkgVersion });
-  const soRes = await savedObjectsClient.bulkGet<PackageAsset>(packageAssets);
-  const paths = soRes.saved_objects.map((asset) => asset.attributes.asset_path);
+  const bulkRes = await savedObjectsClient.bulkGet<PackageAsset>(packageAssets);
+  const entries: ArchiveEntry[] = bulkRes.saved_objects.map((so) => {
+    const { asset_path: path, data_utf8: utf8, data_base64: base64 } = so.attributes;
+    const buffer = utf8 ? Buffer.from(utf8, 'utf8') : Buffer.from(base64, 'base64');
 
-  // create the packageInfo
-  // TODO: this is mostly copied from validtion.ts, but we should save packageInfo somewhere
-  // so we don't need to do this again as this was already done either in registry or through upload
+    if (path && buffer) setArchiveEntry(path, buffer);
```

Comment:
Added this to mirror the approach we take during package installation of adding all items to the cache. We'll want to revisit this as part of a review of how we cache but it's consistent